### PR TITLE
Feature/tmdropdown

### PR DESCRIPTION
**Pull request by ElishaSamPeterPrabhu**
Screenshot_1694501476
Screenshot_1694501470

### CODE REVIEW

File: Trimble.Modus.Components/Styles/Colors.xaml [hunk 1]
```diff
@@ -51,4 +51,7 @@
   <!-- Switch -->
   <Color x:Key="SwitchSelected">#217CBB</Color>
   <Color x:Key="SwitchUnselected">#90939f</Color>
-</ResourceDictionary>
\ No newline at end of file
+  <!--DropDown-->
+  <Color x:Key="DropDownListBorder">#217CBB</Color>
```
Comment (Sofiya-kumar) [inline Trimble.Modus.Components/Styles/Colors.xaml:55]: SwitchSelected color and DropDownListBorder having the same color code. Please refactor this

Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Styles/Colors.xaml:55]: The refactoring of color codes will be done after pushing the remaining components including this.

File: Trimble.Modus.Components/Controls/DropDown/TMDropDown.cs [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+        BindableProperty.Create(nameof(ItemsSource), typeof(IEnumerable), typeof(TMDropDown), null, propertyChanged: OnItemsSourceChanged);
+
+    public static readonly BindableProperty SelectedItemsProperty =
+        BindableProperty.Create(nameof(SelectedItems), typeof(List<object>), typeof(TMDropDown));
+    #endregion
+    #region Constructor
+    public TMDropDown()
+    {
+        SelectedItems = new List<object> { };
+        label = new Label
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/DropDown/TMDropDown.cs:69]: Can't you have all these in the .xaml file?

File: Trimble.Modus.Components/Controls/DropDown/TMDropDown.cs [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+        await Task.WhenAll(
+
+            indicatorButton.RotateTo(0, AnimationDuration)
+        );
+        isVisible = false;
+    }
+
+    private async void Open()
+    {
+        var dataTemplate = new DataTemplate(() =>
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/DropDown/TMDropDown.cs:189]: It looks like you are creating all these UI components whenever the Open methods gets called. Can't we create these once and reuse again?

File: Trimble.Modus.Components/Styles/Colors.xaml (hunk 1 above)
Comment (rthanga1) [reply Trimble.Modus.Components/Styles/Colors.xaml:55]: Yeah, there's a whole lot changes need to be done for the Colors. We have talked about this today. Let's cover this as part of that ticket

**Review by rthanga1 — CHANGES_REQUESTED** [review]

File: Trimble.Modus.Components/Controls/DropDown/TMDropDown.cs (hunk 3 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/DropDown/TMDropDown.cs]: For now components are recreated in open only in windows

File: Trimble.Modus.Components/Controls/DropDown/DropDownContents.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
+using CommunityToolkit.Mvvm.DependencyInjection;
+using System.Collections;
+using Trimble.Modus.Components.Enums;
+using Trimble.Modus.Components.Popup.Animations;
+
+namespace Trimble.Modus.Components;
+
+public partial class DropDownContents : PopupPage
+{
+	public DropDownContents(View anchorView,Enums.ModalPosition position, Thickness margin,IEnumerable itemSource,double desiredHeight,double widthRequest,EventHandler<SelectedItemChangedEventArgs> eventHandler,int selectedIndex,double Y,double height): base(anchorView, position)
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/DropDown/DropDownContents.xaml.cs:10]: There should be a limit for the number of parameters passed. 3 or 4 parameters is okay, when it goes beyod that, method should be optimized.

Comment (rthanga1) [reply Trimble.Modus.Components/Controls/DropDown/DropDownContents.xaml.cs:10]: I have optimized it

**Review by rthanga1 — APPROVED** [review]

**Comment by rthanga1** [comment]
I have fixed the alignment issues in iOS. Please verify Android and Windows once, we can merge. Please resolve the conflicts too

**Comment by ElishaSamPeterPrabhu** [comment]
> I have fixed the alignment issues in iOS. Please verify Android and Windows once, we can merge. Please resolve the conflicts too
Okay @rthanga1

**Review by rthanga1 — APPROVED** [review]

**Review by rthanga1 — APPROVED** [review]

**Pull request merged**